plugin/src/main/java/jenkins/plugins/openstack/compute/JCloudsSlaveTemplate.java

```diff
@@ -153,6 +155,9 @@ private Object readResolve() {
             } else if ("SSH".equals(slaveOptions.slaveType)) {
                 lf = new LauncherFactory.SSH(slaveOptions.credentialsId);
                 slaveOptions.slaveType = slaveOptions.credentialsId = null;
+            } else if ("KAFKA".equals(slaveOptions.slaveType)) {
```

Comment:
There is no need to add these into `readResolve` methods. They are only used to migrate old config to new one. And since kafka is just being added, it does not exist in user configs.